app/components/layout.js

```diff
@@ -19,14 +19,18 @@ class Layout extends React.Component {
     this.props.dispatch(connectToServer(ip));
   }
 
-  handshake(handshakeData) {
-    this.props.dispatch(handshake(handshakeData));
+  handshake() {
+    this.props.dispatch(handshake());
+  }
+
+  cpuProfiling() {
+    this.props.dispatch(startCpuProfiling());
   }
 
   render() {
-    return(
+    return (
       <div>
-        <Navbar {...this.props} connect={this.connect}/>
+        <Navbar {...this.props} connect={::this.connect}/>
```

Comment:
yes similar to `this.connect.bind(this)`. But as we discussed in chatroom we won't be using this any more. I also remove this is some other commit.